caml_z.c

```diff
@@ -695,92 +696,222 @@ CAMLprim value ml_z_of_substring_base(value b, value v, value offset, value leng
   CAMLreturn(r);
 }
 
-CAMLprim value ml_z_to_int(value v)
+/* either stores the result in r and returns 0 (no overflow),
+   or returns 1 and leave r undefined (overflow)
+*/
+static value ml_to_int(value v, value* r)
```

Comment:
Just to clarify the intent, this should be
```suggestion
static int ml_to_int(value v, intnat* r)
```